testnet/stacks-node/src/tests/epoch_205.rs

```diff
@@ -913,3 +916,292 @@ fn test_cost_limit_switch_version205() {
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    eprintln!("expect epoch transition");
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    eprintln!("expect microblock stream");
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    // this test can sometimes miss a mine block event.
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let blocks = test_observer::get_blocks();
+    assert!(blocks.len() >= 5, "Should have produced at least 5 blocks");
+
+    let mut max_big_txs_per_microblock_20 = 0;
+    let mut total_big_txs_per_microblock_20 = 0;
+
+    let mut max_big_txs_per_microblock_205 = 0;
+    let mut total_big_txs_per_microblock_205 = 0;
+
+    let mut in_205 = false;
+
+    // NOTE: this only counts the number of txs per stream, not in each microblock
+    for block in blocks {
+        let transactions = block.get("transactions").unwrap().as_array().unwrap();
+        eprintln!("{}", transactions.len());
+
+        let mut num_big_microblock_txs = 0;
+
+        for tx in transactions.iter() {
+            let raw_tx = tx.get("raw_tx").unwrap().as_str().unwrap();
+            if raw_tx == "0x00" {
+                continue;
+            }
+            let tx_bytes = hex_bytes(&raw_tx[2..]).unwrap();
+            let parsed = StacksTransaction::consensus_deserialize(&mut &tx_bytes[..]).unwrap();
+            if let TransactionPayload::SmartContract(tsc) = parsed.payload {
+                if tsc.name.to_string().find("costs-2").is_some() {
+                    in_205 = true;
+                }
+                if tsc.name.to_string().find("large").is_some() {
+                    if in_205 {
+                        num_big_microblock_txs += 1;
+                        total_big_txs_per_microblock_205 += 1;
+                    } else {
+                        num_big_microblock_txs += 1;
+                        total_big_txs_per_microblock_20 += 1;
+                    }
+                }
+            }
+        }
+        if in_205 && num_big_microblock_txs > max_big_txs_per_microblock_205 {
+            max_big_txs_per_microblock_205 = num_big_microblock_txs;
+        }
+        if !in_205 && num_big_microblock_txs > max_big_txs_per_microblock_20 {
+            max_big_txs_per_microblock_20 = num_big_microblock_txs;
+        }
+    }
+
+    eprintln!(
+        "max_big_txs_per_microblock_20: {}, total_big_txs_per_microblock_20: {}",
+        max_big_txs_per_microblock_20, total_big_txs_per_microblock_20
+    );
+    eprintln!(
+        "max_big_txs_per_microblock_205: {}, total_big_txs_per_microblock_205: {}",
+        max_big_txs_per_microblock_205, total_big_txs_per_microblock_205
+    );
+
+    assert!(max_big_txs_per_microblock_20 < 2);
```

Comment:
interesting that the numbers are `2` and `18`.. you only made the difference in limit 2x, but the difference in trandactions is > 9x.

 it's optoinal but you could pick a smaller number of transactions or a bigger block limit, so that there is some correspondence between the two

Reply:
That's because the cost functions get applied at the epoch transition, which means that the previously-expensive `index-of` Clarity function gets a *lot* cheaper :)